tools/MarkdownConverter.Tests/markdown-lint.xml

```diff
@@ -20,7 +20,10 @@
       <t space="preserve">Note</t>
     </r>
     <r>
-      <t space="preserve">: Note 1. </t>
+      <t space="preserve">: Note 1.</t>
+    </r>
+    <r>
+      <t space="preserve"> </t>
```

Comment:
These are the diffs in output that are causing the unit tests to fail. The updated XML does match the generated output, but for some reason, the difference parser is viewing the new nodes as "null text" rather than a single space.

Reply:
@jskeet 

In my final two commits, I fix the unit tests, by modifying the comparison tests. Only add a failed comparison when the expect node is null and the actual node is a single space.